src/commands/lambda/instrument.ts

```diff
@@ -256,23 +302,29 @@ export class InstrumentCommand extends Command {
 
     const willUpdate = willUpdateFunctionConfigs(configList)
     if (this.interactive && willUpdate) {
-      this.context.stdout.write(`${yellow('[!]')} Confirmation needed.\n`)
+      this.context.stdout.write(renderConfirmationNeededSoftWarning())
       const isConfirmed = await requestChangesConfirmation('Do you want to apply the changes?')
       if (!isConfirmed) {
         return 0
       }
-      this.context.stdout.write(`${yellow('[!]')} Instrumenting functions.\n`)
+      this.context.stdout.write(renderInstrumentingFunctionsSoftWarning())
     }
 
-    const promises = Object.values(configGroups).map((group) =>
-      updateLambdaFunctionConfigs(group.lambda, group.cloudWatchLogs, group.configs)
-    )
-    try {
-      await Promise.all(promises)
-    } catch (err) {
-      this.context.stdout.write(`${red('[Error]')} Failure during update. ${err}\n`)
+    if (willUpdate) {
+      const promises = Object.values(configGroups).map((group) =>
+        updateLambdaFunctionConfigs(group.lambda, group.cloudWatchLogs, group.configs)
+      )
+      const spinner = updatingFunctionsSpinner(promises.length)
+      spinner.start()
+      try {
+        await Promise.all(promises)
```

Comment:
The current `@types/node` wouldn't allow me to use `Promise.allSettled` let's raise a task for this behavior, I'll need to address it in another PR, since we'd have to require the CI package owners to approve the PR for the necessary bump.

Reply:
Closing since I merged #695, will update